core/services/pipeline/orm.go

```diff
@@ -144,10 +144,10 @@ func (o *orm) CreateRun(ctx context.Context, jobID int32, meta map[string]interf
 
 		err = tx.Raw(`
             INSERT INTO pipeline_runs (pipeline_spec_id, meta, created_at)
-            SELECT pipeline_spec_id, $1, NOW()
-            FROM jobs WHERE id = $2
+            SELECT pipeline_spec_id, ?, NOW()
+            FROM jobs WHERE id = ? 
             RETURNING *`, JSONSerializable{Val: meta}, jobID).Scan(&run).Error
-		if gorm.IsRecordNotFoundError(err) {
+		if run.ID == 0 {
```

Comment:
Was this broken before?

Reply:
no its just that gorm v2 only returns record not found errors in a few specific cases now https://gorm.io/docs/v2_release_note.html#ErrRecordNotFound